src/lib/utils/styling/styler.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Inject, Injectable, Optional, PLATFORM_ID} from '@angular/core';
+import {isPlatformBrowser, isPlatformServer} from '@angular/common';
+
+import {applyCssPrefixes} from '../auto-prefixer';
+import {ServerStylesheet} from './server-stylesheet';
+import {SERVER_TOKEN} from './server-token';
+
+@Injectable()
+export class StyleService {
+
+  constructor(@Optional() private _serverStylesheet: ServerStylesheet,
+              @Optional() @Inject(SERVER_TOKEN) private _serverModuleLoaded: boolean,
+              @Inject(PLATFORM_ID) private _platformId) {}
+
+  /**
+   * Applies styles given via string pair or object map to the directive element
+   */
+  applyStyleToElement(element: HTMLElement, style: StyleDefinition, value?: string | number) {
+    let styles = {};
+    if (typeof style === 'string') {
+      styles[style] = value;
+      style = styles;
+    }
+    styles = applyCssPrefixes(style);
+    this._applyMultiValueStyleToElement(styles, element);
+  }
+
+  /**
+   * Applies styles given via string pair or object map to the directive's element
+   */
+  applyStyleToElements(style: StyleDefinition, elements: HTMLElement[] = []) {
+    const styles = applyCssPrefixes(style);
+    elements.forEach(el => {
+      this._applyMultiValueStyleToElement(styles, el);
+    });
+  }
+
+  /**
+   * Determine the DOM element's Flexbox flow (flex-direction)
+   *
+   * Check inline style first then check computed (stylesheet) style
+   */
+  getFlowDirection(target: HTMLElement): [string, string] {
+    const query = 'flex-direction';
+    let value = this.lookupStyle(target, query);
+    if (value === FALLBACK_STYLE) {
+      value = '';
+    }
+    const hasInlineValue = this.lookupInlineStyle(target, query) ||
+    (isPlatformServer(this._platformId) && this._serverModuleLoaded) ? value : '';
+
+    return [value || 'row', hasInlineValue];
+  }
+
+  /**
+   * Find the DOM element's raw attribute value (if any)
+   */
+  lookupAttributeValue(element: HTMLElement, attribute: string): string {
+    return element.getAttribute(attribute) || '';
+  }
+
+  /**
+   * Find the DOM element's inline style value (if any)
+   */
+  lookupInlineStyle(element: HTMLElement, styleName: string): string {
+    return element.style[styleName] || element.style.getPropertyValue(styleName);
+  }
+
+  /**
+   * Determine the inline or inherited CSS style
+   * @TODO(CaerusKaru): platform-server has no implementation for getComputedStyle
```

Comment:
What's the action on this TODO? The whole idea of `getComputedStyle` inherently can't work on the server (without being driven by an entire browser rendering engine)

Reply:
Changed to `NOTE`